src/query/interpreter.cpp

```diff
@@ -976,7 +978,7 @@ struct PullPlan {
   explicit PullPlan(std::shared_ptr<CachedPlan> plan, const Parameters &parameters, bool is_profile_query,
                     DbAccessor *dba, InterpreterContext *interpreter_context, utils::MemoryResource *execution_memory,
                     std::optional<std::string> username, TriggerContextCollector *trigger_context_collector = nullptr,
-                    std::optional<size_t> memory_limit = {});
+                    std::optional<size_t> memory_limit = {}, bool contains_csv = false);
```

Comment:
I would phrase this differently: from the `PullPlan` perspective it doesn't matter if it is a CSV or not: the only information it needs is whether to use the monotonic buffer or not. I would focus on that: `use_monotonic_memory`.